modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java

```diff
@@ -178,21 +174,32 @@ public Driver getJdbcDriverInstance() {
      * @return a Connection
      * @throws SQLException if there is a repeated failure to create the connection
      */
-    public Connection createConnection(String queryString) throws SQLException {
+    public Connection createConnection(String queryString) throws SQLException, NoDriverFoundException {
         final Properties info = new Properties();
         info.put("user", this.getUsername());
         info.put("password", this.getPassword());
         final String url = constructUrlForConnection(queryString);
 
         final Driver jdbcDriverInstance = getJdbcDriverInstance();
 
+        SQLException lastException = null;
         try {
-            return Unreliables.retryUntilSuccess(getConnectTimeoutSeconds(), TimeUnit.SECONDS, () ->
-                DB_CONNECT_RATE_LIMIT.getWhenReady(() ->
-                    jdbcDriverInstance.connect(url, info)));
-        } catch (Exception e) {
-            throw new SQLException("Could not create new connection", e);
+            long start = System.currentTimeMillis();
+            // give up if we hit the time limit or the container stops running for some reason
+            while (System.currentTimeMillis() < start + (1000 * connectTimeoutSeconds) && isRunning()) {
+                try {
+                    logger().debug("Trying to create JDBC connection using {} to {} with properties: {}", driver.getClass().getName(), url, info);
+
+                    return jdbcDriverInstance.connect(url, info);
+                } catch (SQLException e) {
+                    lastException = e;
+                    Thread.sleep(1000L);
```

Comment:
1s is a bit too much, maybe ~100ms?

Reply:
Sure, this is what it was previously, as was the no-delay used above. But I'll set a 100ms delay for both.